model_analyzer/config/generate/neighborhood.py

```diff
@@ -214,48 +214,82 @@ def get_nearest_neighbor(self, coordinate_in: Coordinate) -> Coordinate:
         return nearest_neighbor
 
     def _create_neighborhood(self) -> List[Coordinate]:
+        """
+        Create and return a neighborhood of all Coordinates within 
+        range <_radius> that are also within all bounds
+        """
 
-        neighborhood = []
-        potential_neighborhood = self._get_potential_neighborhood(
-            self._home_coordinate, self._radius)
+        def _is_in_bounds(potential_coordinate: List[int]) -> bool:
+            for i, v in enumerate(potential_coordinate):
+                dim = self._config.get_dimension(i)
+                if (v > dim.get_max_idx() or v < dim.get_min_idx()):
+                    return False
+            return True
 
-        for potential_coordinate in potential_neighborhood:
-            distance = Neighborhood.calc_distance(self._home_coordinate,
-                                                  potential_coordinate)
+        neighborhood = []
+        potential_steps = self._get_potential_steps(
+            self._config.get_num_dimensions(), self._radius)
+
+        for potential_step in potential_steps:
+            for (i, v) in enumerate(self._home_coordinate):
+                potential_step[i] += v
+            if (_is_in_bounds(potential_step)):
+                neighborhood.append(Coordinate(potential_step))
+        return neighborhood
 
-            if distance <= self._radius:
-                neighborhood.append(potential_coordinate)
+    def _get_potential_steps(self, num_coordinates: int,
+                             radius: int) -> List[List[int]]:
+        """ 
+        Create and return a list of all possible step vectors that are 
+        within <_radius> distance
+        """
 
-        return neighborhood
+        result_list = []
+        v = [0] * num_coordinates
+        base = [0] * num_coordinates
+
+        def _append_combinations_to_results(curr_val: List[int],
```

Comment:
In my opinion this is less clean/readable. Took me a couple of minutes to figure out that this method is really just 247-249, and 291-292

Reply:
it also gives access to all of the variables in the parent function so that they don't all have to be explicitly passed into all of the child functions